### DISCUSSION TITLE
multi-button keybinds

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#399] [Ideas]
i'm using a microsoft keyboard which has a few keys sending multiple key signals, among which an emoji key sending `Mod+Shift+Ctrl+Alt+Space`.
i was wondering if i could use that key as a modifier to bind specific emoji to different letters, so as a test tried a bind:

```
Mod+Shift+Ctrl+Alt+Space+F9 { spawn "sh" "-c" "notify-send foo baz'"; }
```

however, this got me:

```
       > Error: error loading config
       >     Diagnostic severity: error
       >     Caused by: error parsing
       >     Caused by: error parsing KDL
       >
       > Error: invalid keybind
       >     Diagnostic severity: error
       >
       >     Caused by: invalid keybind
       >     Caused by: invalid modifier: Space
       > Begin snippet for .attr-041r6j70w6v6dakqkzmwb899h71g2hibch7d14bbv9gy7bxhd5mp starting at line 144, column 1
```

so i get that so far binds use mods + one non-mod key. i guess i'm curious if it might be possible to have binds with multiple non-mod keys this way as well.

**Comment by YaLTeR** [comment]
Uhh, not easily at least. Since modifiers are held down and their state is always known, whereas normal keys are not stored the same way. So this might need either Smithay refactors, or some very ugly code to make work I expect.
